src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobRandomPos.java

```diff
@@ -57,11 +62,36 @@ public PathJobRandomPos(
     {
         super(world, start, start, range, new RandomPathResult(), entity);
         this.distance = distance;
+        this.minDistance = range;
 
         final Tuple<Direction, Direction> dir = BlockPosUtil.getRandomDirectionTuple(random);
         this.destination = start.offset(dir.getA(), distance).offset(dir.getB(), distance);
     }
 
+    /**
+     * Prepares the PathJob for the path finding system.
+     *
+     * @param world         world the entity is in.
+     * @param start         starting location.
+     * @param minDistFromSpawn how far to move away.
+     * @param range         max range to search.
+     * @param entity        the entity.
+     */
+    public PathJobRandomPos(
+      final World world,
+      @NotNull final BlockPos start,
+      final int minDistFromSpawn,
+      final int range,
+      final int minDistToDest,
```

Comment:
think this would be a maxDistToDest right?^^

Reply:
maxDistFromDest or minDistToDest